File: src/main/java/org/jesperancinha/xml/adder/XmlAdderMain.java
package org.jesperancinha.xml.adder;

import javax.xml.parsers.ParserConfigurationException;
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;

public class XmlAdderMain {

    public static void main(String[] args) throws IOException, ParserConfigurationException {
        final String sourceDirectory = args[0];
        final String destinationDirectory = args[1];
        final String addAttributesFile = args[2];
        final String deleteAttributesFile = args[3];
        final String rule = args[4];

        final File fileSourceDirectory = new File(sourceDirectory);
        final File fileDestinationDirectory = new File(destinationDirectory);
        final InputStream fileAddAtributesFile = new FileInputStream(new File(addAttributesFile));
        final InputStream fileDeleteAttributesFile = new FileInputStream(new File(deleteAttributesFile)) ;
        final InputStream fileRule = new FileInputStream(new File(rule));

        final XmlAdderManager manager = new XmlAdderManager( //
                fileSourceDirectory, //
                fileDestinationDirectory, //
                fileAddAtributesFile, //
                fileDeleteAttributesFile, //
                fileRule);
    }
}

File: jeorg-kotlin-apps/xml-adder/app/src/main/kotlin/org/jesperancinha/xml/adder/XmlAdderMain.kt
package org.jesperancinha.xml.adder

import java.io.File
import java.io.FileInputStream
import java.io.IOException
import java.io.InputStream
import javax.xml.parsers.ParserConfigurationException

object XmlAdderMain {
    @Throws(IOException::class, ParserConfigurationException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val sourceDirectory = args[0]
        val destinationDirectory = args[1]
        val addAttributesFile = args[2]
        val deleteAttributesFile = args[3]
        val rule = args[4]
        val fileSourceDirectory = File(sourceDirectory)
        val fileDestinationDirectory = File(destinationDirectory)
        val fileAddAtributesFile: InputStream = FileInputStream(File(addAttributesFile))
        val fileDeleteAttributesFile: InputStream = FileInputStream(File(deleteAttributesFile))
        val fileRule: InputStream = FileInputStream(File(rule))
        XmlAdderManager(
            fileSourceDirectory,
            fileDestinationDirectory,
            fileAddAtributesFile,
            fileDeleteAttributesFile,
            fileRule
        )
    }
}